integration-tests/tests/cases/mpc.rs

```diff
@@ -233,3 +234,44 @@ fn sign_arg(seed: u8) -> SignArgs {
         key_version: 0,
     }
 }
+
+/// drop the first 20 presignature messages on each node and see if the system
+/// can recover
+#[tokio::test(flavor = "multi_thread")]
+async fn test_presignature_timeout() {
+    fn create_filter() -> MessageFilter {
+        let mut drop_counter = 20;
+        Box::new(move |(msg, _)| {
+            let pass = match msg {
+                mpc_node::protocol::Message::Presignature(_) => drop_counter == 0,
+                _ => true,
+            };
+
+            if !pass {
+                drop_counter -= 1;
+            }
+            pass
+        })
+    }
+
+    let network = MpcFixtureBuilder::default()
+        // configure network ready to generate presignatures immediately
+        .with_preshared_key()
+        .with_preshared_triples()
+        .with_min_presignatures_stockpile(5)
+        .with_max_presignatures_stockpile(5)
+        .with_min_triples_stockpile(0)
+        .with_max_triples_stockpile(0)
+        // apply message filter to all nodes
+        .with_outgoing_message_filter(0, create_filter())
+        .with_outgoing_message_filter(1, create_filter())
+        .with_outgoing_message_filter(2, create_filter())
+        // speed up timeout
+        .with_presignature_timeout_ms(2000)
+        .build()
+        .await;
+
+    tokio::time::timeout(Duration::from_secs(10), network.wait_for_presignatures(1))
```

Comment:
To be sure I understand this test correctly.
- We start 3 nodes
- They try to generate 5 per-signatures
- Some or all (not clear) of this protocols fail, since we drop 20 per-signature messages on each node
- Nodes start new protocols, at least one of them succeeds.

It means that old protocols are not waiting for new messages forever and the counter of ongoing generators represents protocols that are actually alive.

The requirement for such test is: The number of failed protocols must be >= number of concurrent pre-signature generations. This is not guaranteed by the test itself, but it is very nice to have anyways.